cpp/src/arrow/filesystem/localfs.cc

```diff
@@ -210,6 +211,22 @@ Result<FileInfo> StatFile(const std::string& path) {
   return info;
 }
 
+Result<FileInfo> IdentifyFile(const std::string& path) {
```

Comment:
This is defined within the POSIX branch (else) of the `#ifdef _WIN32` condition above. Since it's platform agnostic, I think you can pull this out to the top-level.